src/current-refinements/current-refinements.ts

```diff
@@ -85,9 +83,31 @@ export class NgAisCurrentRefinements extends BaseWidget {
   }
 
   get refinements() {
-    return isFunction(this.transformItems)
+    const items = isFunction(this.transformItems)
       ? this.transformItems(this.state.refinements)
       : this.state.refinements;
+
+    // group refinements by category? (attributeName && type)
+    return items.reduce((res, { type, attributeName, ...refinement }) => {
+      const match = res.find(
+        r => r.attributeName === attributeName && r.type === type
+      );
+      if (match) {
+        match.items.push({ type, attributeName, ...refinement });
+      } else {
+        res.push({
+          type,
+          attributeName,
+          label: capitalize(attributeName),
+          items: [{ type, attributeName, ...refinement }]
+        });
+      }
+      return res;
+    }, []);
+  }
+
```

Comment:
@samouss ^^^

Reply:
Yep we are doing pretty much the same thing in React InstantSearch. Except that we do it in each connector instead of doing it in `connectCurrentRefinements`. If you want to take a look, all the widgets that apply a refinement have a `getMetadata` function. This function return an object that will pushed in an array and pass to the `connectCurrentRefinements` as `metadata`. 

See the links below for `connectMenu` & `connectCurrentRefinements`:

https://github.com/algolia/react-instantsearch/blob/3da8040319c2fcdc69e620bf80f5d64c3204c75f/packages/react-instantsearch/src/connectors/connectMenu.js#L223-L245

https://github.com/algolia/react-instantsearch/blob/3da8040319c2fcdc69e620bf80f5d64c3204c75f/packages/react-instantsearch/src/connectors/connectCurrentRefinements.js#L23-L52

Reply:
Alright, thanks 💯 

We do not have this API on the connectors, and I cannot easily call a method on every mounted widgets. So I will keep the current logic 👍